Gems/GeoJSONSpawner/Code/Source/GeoJSONSpawner/ROS2Interface/SubscriptionHandler.h

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (C) Robotec AI - All Rights Reserved
+ *
+ * This source code is protected under international copyright law.  All rights
+ * reserved and protected by the copyright holders.
+ * This file is confidential and only available to authorized individuals with the
+ * permission of the copyright holders.  If you encounter this file and do not have
+ * permission, please contact the copyright holders and delete this file.
+ */
+
+#pragma once
+
+#include <AzCore/std/function/function_template.h>
+
+#include <ROS2/RobotControl/ControlSubscriptionHandler.h>
+
+namespace GeoJSONSpawner::ROS2Interface
+{
+
+    //! Auxiliary class that wraps subscriber for a ROS2 topic
+    template<typename MsgType>
+    class SubscriptionHandler : public ROS2::ControlSubscriptionHandler<MsgType>
```

Comment:
This handler doesn't makes sense here. `ControlSubscriptionHandler` is for ROS2 Robots control.

Replace it with standard ROS2 subscribers.

Reply:
I would move all ROS2 stuff to separate Gem. It will make sense for non-robotics usecases.
We will split georeference to other gem eventually.